src/modules/tas_studio/editor/mod.rs

```diff
@@ -2241,6 +2340,73 @@ impl Editor {
         self.apply_operation(op)
     }
 
+    /// Sets yawspeed of the selected frame bulk.
+    pub fn set_yawspeed(&mut self, new_yawspeed: Option<f32>) -> eyre::Result<()> {
+        // Don't toggle during active adjustments for consistency with other operations.
+        if self.is_any_adjustment_active() {
+            return Ok(());
+        }
+
+        if self.in_camera_editor {
+            return Ok(());
+        }
+
+        let Some(bulk_idx) = self.selected_bulk_idx else {
+            return Ok(());
+        };
+        let (line_idx, bulk) = self
+            .branch()
+            .branch
+            .script
+            .lines
+            .iter()
+            .enumerate()
+            .filter_map(|(line_idx, line)| line.frame_bulk().map(|bulk| (line_idx, bulk)))
+            .nth(bulk_idx)
+            .unwrap();
+
+        let mut new_bulk = bulk.clone();
+
+        if let Some(AutoMovement::Strafe(StrafeSettings { type_: _, ref mut dir })) =
+            &mut new_bulk.auto_actions.movement
+        {
+            let new_yawspeed = if let Some(yawspeed) = new_yawspeed {
+                if yawspeed == 0. {
+                    None
+                } else {
+                    new_yawspeed
+                }
+            } else {
+                new_yawspeed
+            };
+
+            match dir {
+                StrafeDir::Left(_) => *dir = StrafeDir::Left(new_yawspeed),
+                StrafeDir::Right(_) => *dir = StrafeDir::Right(new_yawspeed),
+                _ => (),
+            }
+        }
+
+        if new_bulk == *bulk {
+            return Ok(());
+        }
+
+        let mut buffer = Vec::new();
+        hltas::write::gen_frame_bulk(&mut buffer, bulk)
+            .expect("writing to an in-memory buffer should never fail");
+        let from = String::from_utf8(buffer)
+            .expect("FrameBulk serialization should never produce invalid UTF-8");
+
+        let mut buffer = Vec::new();
+        hltas::write::gen_frame_bulk(&mut buffer, &new_bulk)
+            .expect("writing to an in-memory buffer should never fail");
+        let to = String::from_utf8(buffer)
+            .expect("FrameBulk serialization should never produce invalid UTF-8");
+
+        let op = Operation::Replace { line_idx, from, to };
```

Comment:
If we switch to a separate strafing type as I mentioned in the hltas PR, then this function will only need to work on s5x frame bulks, and can use Operation::SetYawSpeed (which will be less taxing on the database size compared to Replace).